lib/vrl/web-playground/public/vrl-highlighter.js

```diff
@@ -0,0 +1,636 @@
+// VRL Language Definition
+export let vrlLanguageDefinition = {
+    defaultToken: "invalid",
+    ignoreCase: true,
+    tokenPostfix: ".vrl",
+
+    brackets: [
+        { open: "{", close: "}", token: "delimiter.curly" },
+        { open: "[", close: "]", token: "delimiter.square" },
+        { open: "(", close: ")", token: "delimiter.parenthesis" },
+    ],
+
+    regEx: /\/(?!\/\/)(?:[^\/\\]|\\.)*\/[igm]*/,
+
+    keywords: [
+        "abort",
+        "as",
+        "break",
+        "continue",
+        "else",
+        "false",
+        "for",
+        "if",
+        "impl",
+        "in",
+        "let",
+        "loop",
+        "null",
+        "return",
+        "self",
+        "std",
+        "then",
+        "this",
+        "true",
+        "type",
+        "until",
+        "use",
+        "while",
+    ],
+
+    // we include these common regular expressions
+    symbols: /[=><!~?&%|+\-*\/\^\.,\:]+/,
+    escapes: /\\(?:[abfnrtv\\"'$]|x[0-9A-Fa-f]{1,4}|u[0-9A-Fa-f]{4}|U[0-9A-Fa-f]{8})/,
+
+    // The main tokenizer for our languages
+    tokenizer: {
+        root: [
+            // function invokes fallible
+            [
+                /([a-zA-Z_!]+)(\!)(\()/,
+                {
+                    cases: {
+                        $3: ['keyword', 'keyword', '']
+                    }
+                    //   log: 'in function invoke::found::\n\n$0'
+                },
+            ],
+            // function invokes
+            [
+                /([a-zA-Z_!]+)(\()/,
+                {
+                    cases: {
+                        $2: ['keyword', '']
+                    }
+                    //   log: 'in function invoke::found::\n\n$0'
+                },
+            ],
+
+            // rstrings
+            [/r'[^']+'/, { token: "regexp" /*log: 'root_r_string::\n\n$0'*/ }],
+
+            // timestamps
+            [/t'[^']+'/, { token: "regexp" /*log: 'function_arg_r_string::\n\n$0'*/ }],
+
+            // field access, eg: .foo
+            [/(\.[^\ \=]+)([\ |\=])/, {
+                cases: {
+                    $2: ["entity", ""]
+                }
+            }],
```

Comment:
This still misses a bunch of possible path syntax styles.  Here's an example of what we do support:

https://deploy-preview-15557--vrl-playground.netlify.app/?state=eyJwcm9ncmFtIjoiLmZvb1xuLihiYXIgfCBiYXopXG4uZm9vLihiYXIgfCBiYXogfCBxdXgpXG4uWzEyXVxuLmZvb1sxXVxuLmZvby4oYmFyIHwgYmF6IHwgcXV4KVsxMl1cbi5mb29bMTJdLmJhci4oYmF6IHwgcXV4KS5xdXV4WzFdIiwiZXZlbnQiOnsibWVzc2FnZSI6IkhlbGxvIFZSTCIsImZvbyI6ImRlbGV0ZSBtZSIsImh0dHBfc3RhdHVzIjoiMjAwIn0sImlzX2pzb25sIjpmYWxzZX0%3D